Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -297,12 +312,60 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
                     {
                         string installationDirectory = GetInstallationDirectory(id);
                         localPackage = await InstallExtension(repoPackage, installationDirectory, feedUrl);
-                        GetSettingManager(id).SetValue(_feedUrlSetting, feedUrl);
+                        siteExtensionSettings.SetValues(new KeyValuePair<string, JToken>[] {
+                            new KeyValuePair<string, JToken>(_versionSetting, localPackage.Identity.Version.ToNormalizedString()),
+                            new KeyValuePair<string, JToken>(_feedUrlSetting, feedUrl),
+                            new KeyValuePair<string, JToken>(_installUtcTimestampSetting, DateTime.UtcNow.ToString("u"))
+                        });
                     }
+
+                    return new Tuple<bool, SiteExtensionInfo>(true, await ConvertLocalPackageToSiteExtensionInfo(localPackage, checkLatest: true));
                 }
 
-                return await ConvertLocalPackageToSiteExtensionInfo(localPackage, checkLatest: true);
+                return new Tuple<bool, SiteExtensionInfo>(false, null);
+            }
+        }
+
+        /// <summary>
+        /// <para> Return true if any of below cases is satisfied:</para>
+        /// <para>      1) Package with same version and from same feed already exist in local repo</para>
+        /// <para>      2) If given feedUrl is null</para>
+        /// <para>              Try to use feed from local package, if feed from local package also null, fallback to default feed</para>
+        /// <para>              Check if version from query is same as local package</para>
+        /// <para>      3) If given version and feedUrl are null</para>
+        /// <para>              Try to use feed from local package, if feed from local package also null, fallback to default feed</para>
+        /// <para>              Check if version from query is same as local package</para>
+        /// </summary>
+        private async Task<bool> IsSiteExtensionInstalled(string id, string version, string feedUrl)
```

Comment:
right now actually most of the case feedUrl is null.
when install site extension from SCM site, feedUrl is always null and relying on code to fallback to "siteextensions.net"